javascriptv3/example_code/iam/scenarios/src/iam_basics.js

```diff
@@ -0,0 +1,410 @@
+node iam_basics.js <user name> <s3 policy name> <role name> <assume policy name>
+
+Running the code:
+node iam_basics.js
+*/
+// snippet-start:[javascript.iam_scenarios.iam_basics]
+// Import required AWS SDK clients and commands for Node.js.
+import { iamClient, REGION } from "../libs/iamClient.js"; // Helper function that creates an Amazon IAM service client module.
+import {
+  CreateUserCommand,
+  CreateAccessKeyCommand,
+  CreatePolicyCommand,
+  CreateRoleCommand,
+  AttachRolePolicyCommand,
+  AttachUserPolicyCommand,
+  DeleteAccessKeyCommand,
+  DeleteUserCommand,
+  DeleteRoleCommand,
+  DeletePolicyCommand,
+  DetachUserPolicyCommand,
+  DetachRolePolicyCommand,
+} from "@aws-sdk/client-iam";
+import { ListBucketsCommand, S3Client } from "@aws-sdk/client-s3";
+import { AssumeRoleCommand, STSClient } from "@aws-sdk/client-sts";
+
+if (process.argv.length < 6) {
+  console.log(
+      "Usage: node iam_basics.js <user name> <s3 policy name> <role name> <assume policy name>\n" +
+      "Example: node iam_basics.js test-user my-s3-policy my-iam-role my-assume-role"
+  );
+}
+// Set the parameters.
+const region = REGION;
+const userName = process.argv[2];
+const s3_policy_name = process.argv[3];
+const role_name = process.argv[4];
+const assume_policy_name = process.argv[5];
+
+// Helper function to delay code execution while AWS Service calls wait for responses.
+function wait(ms) {
+  var start = Date.now();
+  var end = start
+  while (end < start + ms){
+    end = Date.now()
+  }
+}
+
+export const run = async (
+    userName,
+    s3_policy_name,
+    role_name,
+    assume_policy_name
+) => {
+  try {
+    // Create a new user.
+    const user_params = { UserName: userName };
+    console.log("\nCreating a user name " + user_params.UserName + "...\n");
+    const data = await iamClient.send(
+        new CreateUserCommand({ UserName: userName })
+    );
+    const user_arn = data.User.Arn;
+    const user_name = data.User.UserName;
+    console.log(
+        "User with name" + user_name + " and ARN " + user_arn + " created."
+    );
+    try {
+      // Create access keys for the new user.
+      console.log(
+          "\nCreating access keys for " + user_params.UserName + "...\n"
+      );
+      const access_key_params = { UserName: user_name };
+      const data = await iamClient.send(
+          new CreateAccessKeyCommand(access_key_params)
+      );
+      console.log("Success. Access key created: ", data.AccessKey.AccessKeyId);
+      var myAccessKey = data.AccessKey.AccessKeyId;
+      var mySecretAccessKey = data.AccessKey.SecretAccessKey;
+
+      try {
+        // Attempt to list S3 buckets.
+        console.log(
+            "\nWaiting 10 seconds for user and access keys to be created...\n"
+        );
+        wait(10000);
+        console.log(
+            "Attempt to list S3 buckets with the new user (without permissions)...\n"
+        );
+        // Use the credentials for the new user you created.
```

Comment:
// Use the credentials for the new user that you created.

Reply:
done